wireapp/wire-ios-mono
# feat(backup): [DO NOT MERGE] support importing multiplatform backups [WPB-11230]

### Issue

We have different backup formats on different platforms. Not great for users.

The goal is to have a single format for all platforms, and in the future we can use this backup to sync history between clients. Like client A sending its history to client B when logging in, etc.

### Important notes

1. The main goal of this PR is not to be merged.
2. The multiplatform backup library is a **work in progress**.
3. The goal is to get feedback and showcase its usage.
4. Encryption/Decryption is being developed

Because of **[2]**, the `backup.xcframework` is not properly published to a regular dependency repository, but rather being added to the project just to get quick feedback before actually setting up automated deployment.

### Testing

1. Download this file: https://github.com/user-attachments/assets/db06c630-153b-41c1-b533-8755babaa0c1
3. Rename it to "something.ios_wbu" (suffix needed for now, GitHub doesn't allow all formats).
6. Login on the iOS app
7. Import this file as a backup
8. Use an _empty_ password. (Encryption is WIP)
9. See the console:

![image](https://github.com/user-attachments/assets/292349f0-33fa-436f-8762-4776784e6d8b)


---

### Checklist


---
